cpp/lib/src/MarcWriter.cc

```diff
@@ -0,0 +1,158 @@
+#include "MarcWriter.h"
+#include "util.h"
+
+//
+// Marc 21 stuff:
+//
+
+static const size_t MAX_MARC_21_RECORD_LENGTH = 99999;
+static char write_buffer[MAX_MARC_21_RECORD_LENGTH];
+
+
+static bool inline hasDirectoryEntryEnoughSpace(const size_t baseAddress, const size_t data_length, const size_t next_field_length) {
```

Comment:
Also, regular functions (like static member functions) should start with an uppercase letter.